yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1183,6 +1187,29 @@ object Client extends Logging {
     YarnSparkHadoopUtil.addPathToEnvironment(env, Environment.CLASSPATH.name, path)
 
   /**
+   * Returns the path to be sent to the NM for a local path string.
+   *
+   * This method uses two configuration values:
+   *
+   * - spark.yarn.config.localPath: a string that identifies a portion of the input path that may
+   *   only be valid in the local process.
+   * - spark.yarn.config.clusterPath: a string with which to replace the local path. This may
+   *   contain, for example, env variable references, which will be expanded by the NMs when
+   *   starting containers.
+   *
+   * If either config is not available, the input path is returned.
+   */
+  def getClusterPath(conf: SparkConf, path: String): String = {
+    val localPath = conf.get("spark.yarn.config.localPath", null)
+    val clusterPath = conf.get("spark.yarn.config.clusterPath", null)
+    if (localPath != null && clusterPath != null) {
+      path.replace(localPath, clusterPath)
```

Comment:
do you think it makes sense to replace anywhere in the path?  maybe it would be better to only replace a prefix?  I know this is more general, but I'm just worried it might do some inadvertent replacement.

(if we really want it to be general, we could use a regex ... but I feel like simpler is probably better)